http/jsonapi/generator/generate_helper.go

```diff
@@ -23,76 +23,98 @@ func (g *Generator) addGoDoc(typeName, description string) {
 	}
 }
 
-func (g *Generator) goType(stmt *jen.Statement, schema *openapi3.Schema, tags map[string]string) error { // nolint: gocyclo
-	switch schema.Type {
+func (g *Generator) goType(stmt *jen.Statement, schema *openapi3.Schema, tags map[string]string) *typeGenerator { // nolint: gocyclo
+	return &typeGenerator{
+		g:      g,
+		stmt:   stmt,
+		schema: schema,
+		tags:   tags,
+	}
+}
+
+type typeGenerator struct {
```

Comment:
a test is missing for the generator, could you please change one of the generator/internal packages and adopt a test so see that it works.

Reply:
done